### PR TITLE
Student: submit responses: allow student to add a comment for each response(rank, constsum and numerical scale questions) #7239

### LINKED ISSUE #7239: Student: submit responses: allow student to add a comment for each response
When students submit responses, it is useful if the system allows student to add an explanatory comment. It also avoid the need to add questions such as 'explain your response to the previous question' to collect justifications.

This might have to be done as multiple PRs, one per question type. We can start with MCQ questions.

### PR DESCRIPTION
**Pull request by Mynk96**
Fixes #7239 partially

### CODE REVIEW

**Comment by Mynk96** [comment]
@chowyb Please Review!

**Comment by Mynk96** [comment]
UI changes:-

Initially:-
errorrrrrr





Final solution:-
constsum sol

**Comment by Mynk96** [comment]
@chowyb I also tried to reproduce this error but wasn't able to. What I think would have happened is that the `response` was fetched before it had got saved in the `Database`. As a precautionary measure, I have made slight changes and also removed a redundant comment.Please Review!

**Comment by Mynk96** [comment]
@chowyb Yes, just pushing the `resolved merge conflict` commit

**Comment by Mynk96** [comment]
@chowyb Please Review!

File: src/test/java/teammates/test/cases/browsertests/InstructorFeedbackSubmitPageUiTest.java [hunk 1]
```diff
@@ -299,6 +367,186 @@ private void testSubmitAction() throws Exception {
...
+        submitPage.addFeedbackResponseComment("-1-1-19", "New Constsum Comment 6");
+        submitPage.addFeedbackResponseComment("-2-1-19", "New Constsum Comment 7");
+        submitPage.addFeedbackResponseComment("-0-1-20", "New Constsum Comment 8");
+        submitPage.addFeedbackResponseComment("-1-1-20", "New Constsum Comment 9");
+        submitPage.addFeedbackResponseComment("-2-1-20", "New Constsum Comment 10");
+        submitPage.addFeedbackResponseComment("-0-1-23", "New Rank(Recipients) Comment 1");
+        submitPage.addFeedbackResponseComment("-1-1-23", "New Rank(Recipients) Comment 2");
+        submitPage.addFeedbackResponseComment("-2-1-23", "New Rank(Recipients) Comment 3");
+        submitPage.addFeedbackResponseComment("-3-1-23", "New Rank(Recipients) Comment 4");
+        submitPage.addFeedbackResponseComment("-0-1-24", "New Rank(Options) Comment 5");
```
Comment (chowyb) [inline src/test/java/teammates/test/cases/browsertests/InstructorFeedbackSubmitPageUiTest.java]: Should probably be "Comment 1"

Comment (Mynk96) [inline src/test/java/teammates/test/cases/browsertests/InstructorFeedbackSubmitPageUiTest.java]: My bad. I'll change.

**Comment by chowyb** [comment]
@Mynk96 seems okay (build passing), I'll change the branch now.

**Comment by Mynk96** [comment]
@chowyb Please assign other devs to review this branch.

**Comment by Mynk96** [comment]
@chowyb Should I create a new PR with all these changes?

**Comment by chowyb** [comment]
@Mynk96 yes, create one final PR from the newly updated `add-student-response-comment`. Be sure to open the PR from a branch on your own fork, as you will most likely have to make changes as requested by other reviewers.

**Pull request merged**